fendermint/ipc/src/cache.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use num_traits::{PrimInt};
+use std::cmp::{max, min};
+use std::collections::HashMap;
+use std::hash::Hash;
+
+type Bounds<T> = (T, T);
+
+/// The key value cache. The key must be numeric and falls within a range/bound.
+pub struct RangeKeyCache<Key, Value> {
```

Comment:
```suggestion
pub struct RangeKeyCache<K, V> {
```

I think these won't be harder to read, and are well established?